modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java

```diff
@@ -532,6 +535,37 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
         }
     }
 
+    @Override
+    public CodegenProperty fromProperty(String name, Schema p) {
+        final CodegenProperty property = super.fromProperty(name, p);
+
+        // Handle composed properties
+        if (ModelUtils.isComposedSchema(p)) {
+            ComposedSchema composed = (ComposedSchema) p;
+
+            // Count the occurrences of allOf/anyOf/oneOf with exactly one child element
+            long count = Stream.of(composed.getAllOf(), composed.getAnyOf(), composed.getOneOf())
```

Comment:
@kuhnroyal thanks again for the PR. I now recall there's similar change before to another generator do something very similar but some users do not like it.

Let me know when you're free in the coming week and we can discuss further via Slack.